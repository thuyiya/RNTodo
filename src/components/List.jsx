import React, { useRef } from "react";
import { ScrollView } from "react-native";

const List = ({ children }) => {
  const scrollView = useRef();

  return (
    <ScrollView
      style={{ flex: 1 }}
      ref={(ref) => (scrollView.current = ref)}
      onContentChange={() => {
        scrollView.current.scrollToEnd({ animated: true });
      }}
      showsVerticalScrollIndicator={false}
    >
        {children}
    </ScrollView>
  );
};

export default List;
