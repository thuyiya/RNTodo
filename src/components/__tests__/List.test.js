import React from "react";
import renderer from "react-test-renderer";
import { Text } from 'react-native';
import List from "../List";

describe("<List />", () => {
  it("has empty children", () => {
    const tree = renderer.create(<List></List>).toJSON();
    expect(tree.children.length).toBe(1);
  });

  it("has 1 children", () => {
    const tree = renderer.create(<List>
        <Text>Hi 1</Text>
    </List>).toJSON();
    expect(tree.children[0].children.length).toBe(1);
  });

  it("wrap with ScrollView", () => {
    const tree = renderer.create(<List></List>).toJSON();
    expect(tree.type).toBe("RCTScrollView");
  });

  it("renders correctly", () => {
    const tree = renderer.create(<List></List>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
