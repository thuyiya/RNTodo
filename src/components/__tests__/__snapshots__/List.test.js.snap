// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders correctly 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
  style={
    {
      "flex": 1,
    }
  }
>
  <View />
</RCTScrollView>
`;
